base/collection.go

```diff
@@ -140,7 +140,13 @@ func GetGocbV2BucketFromCluster(cluster *gocb.Cluster, spec BucketSpec, waitUnti
 		clusterCompatMajorVersion: uint64(clusterCompatMajor),
 		clusterCompatMinorVersion: uint64(clusterCompatMinor),
 	}
+	uuid, err := getServerUUID(gocbv2Bucket)
+	if err != nil {
+		_ = cluster.Close(&gocb.ClusterCloseOptions{})
+		return nil, fmt.Errorf("%s", err)
```

Comment:
Avoid error wrapping without `%w` or just return err so callers can continue to do errors.Is/As handling 

```suggestion
		return nil, fmt.Errorf("Couldn't get server UUID: %w", err)
```

or

```suggestion
		return nil, err
```